tests/net/arp/src/main.c

```diff
@@ -176,12 +176,14 @@ static inline struct net_pkt *prepare_arp_reply(struct net_if *iface,
 	struct net_eth_hdr *eth;
 
 	pkt = net_pkt_alloc_with_buffer(iface, sizeof(struct net_eth_hdr) +
+					sizeof(struct net_eth_hdr) +
```

Comment:
Having extra ethernet header here is indeed suspicious. After looking the code, the reason for this is that the test creates the reply packet to the same buffer as where the incoming ARP is in. That is why there is two `net_buf_pull(pkt->buffer, sizeof(struct net_eth_hdr));` calls in that function. Not sure if this is worth changing, the code seems to work ok although is a bit weird looking.

Reply:
I thought it was worth changing, but I wasn't able to make the test pass so I must be missing some detail.